tests/RazorLight.Tests/IntegrationTests/RendererCommonCasesTests.cs

```diff
@@ -0,0 +1,111 @@
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Snapper;
+using Snapper.Attributes;
+using Xunit;
+
+namespace RazorLight.Tests.IntegrationTests
+{
+	public class TestViewModel
+	{
+		public string Name { get; set; }
+
+		public int NumberOfItems { get; set; }
+	}
+	
+	[UpdateSnapshots]
+	public class RendererCommonCasesTests
+	{
+
+		[Fact]
+		public async Task Should_Render_Section_And_ViewModel()
+		{
+			var path = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location); 
+			
+			var engine = new RazorLightEngineBuilder()
+					.UseFileSystemProject(Path.Combine(path, "Assets", "Files"))
+					.Build();
+
+			var model = new TestViewModel {Name = "RazorLight", NumberOfItems = 100};
+			var renderedResult = await engine.CompileRenderAsync("template4.cshtml", model);
+			Assert.NotNull(renderedResult);
+			renderedResult.ShouldMatchSnapshot();
+		}
+		
+		[Fact]
+		public async Task Should_Render_Sections_With_IncludeAsync()
+		{
+			var path = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location); 
+			
+			var engine = new RazorLightEngineBuilder()
+				.UseFileSystemProject(Path.Combine(path, "Assets", "Files"))
+				.Build();
+
+			var model = new TestViewModel {Name = "RazorLight", NumberOfItems = 200 };
+			var renderedResult = await engine.CompileRenderAsync("template5.cshtml", model);
+			Assert.NotNull(renderedResult);
+			renderedResult.ShouldMatchSnapshot();
+		}		
+		
+		[Fact]
+		public async Task Should_Fail_When_Required_Section_Is_Missing()
+		{
+			var path = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location); 
+			
+			var engine = new RazorLightEngineBuilder()
+				.UseFileSystemProject(Path.Combine(path, "Assets", "Files"))
+				.Build();
+
+			var model = new TestViewModel {Name = "RazorLight", NumberOfItems = 300};
+			var renderedResult = await engine.CompileRenderAsync("template6.cshtml", model);
```

Comment:
I believe this line is wrong.  The method should be:

```c#
		[Fact]
		public async Task Should_Fail_When_Required_Section_Is_Missing()
		{
			var path = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location); 
			
			var engine = new RazorLightEngineBuilder()
				.UseFileSystemProject(Path.Combine(path, "Assets", "Files"))
				.Build();

			var model = new TestViewModel {Name = "RazorLight", NumberOfItems = 300};
			Assert.ThrowsAsync<TemplateCompilationException>(async () => await engine.CompileRenderAsync("template6.cshtml", model)) ;
			//Assert.NotNull(renderedResult);
			//renderedResult.ShouldMatchSnapshot();
		}
```

If you agree, please update PR.

Reply:
Oh sure, that looks alright somehow I thought it would just not render the sections when they are missing.